stlab/concurrency/utility.hpp

```diff
@@ -99,32 +118,110 @@ T blocking_get(future<T> x) {
     return std::move(*result);
 }
 
+template <typename T>
+stlab::optional<T> blocking_get(future<T> x, const std::chrono::nanoseconds& timeout) {
+    std::exception_ptr error = nullptr;
+    auto state = std::make_shared<detail::shared_state_result<T>>();
+    auto hold = std::move(x).recover(immediate_executor, [_weak_state = make_weak_ptr(state)](auto&& r) {
+        auto state = _weak_state.lock();
+        if (!state) {
+            return;
+        }
+        if (r.error())
```

Comment:
There is an asymmetry between both versions: This accesses state->error not under the lock, but the next one does. Hmm.

Reply:
No need to access the error under a lock.